server/src/main/java/org/elasticsearch/search/query/QuerySearchResult.java

```diff
@@ -366,7 +371,11 @@ public void writeToNoId(StreamOutput out) throws IOException {
             out.writeBoolean(false);
         } else {
             out.writeBoolean(true);
-            aggregations.writeTo(out);
+            if (out.getVersion().before(Version.V_8_0_0)) {
+                aggregations.get().writeTo(out);
```

Comment:
we can maybe get the aggs once if the remote node is in a version before v8 (instead of calling get here and below to get the pipeline aggs) ?

Reply:
@nik9000 ? should we avoid the double deserialization if we need the pipeline aggs below ?

Reply:
Darn it. I twisted the other side around but missed this comment. Of course!

Reply:
Mostly these are going to be the "referencing" ones anyway. But I'll turn it around.